stm32 DMA driver has macro to help channel configuration

FRASTM opened the pull request:
This PR adds a include/zephyr/dt-bindings/dma/stm32_dma.h header file with all the bit fields for the
stm32 DMA channel configuration.
Makes the dma channel config more readable
from `<&dma2 7 5 0x440 0x03>`
to `<&dma2 7 5 DMA_STM32_PERIPH_TX_DEFAULT DMA_STM32_FIFO_FULL>`

User can choose a combination of ORed values to write the bitfield, like : 
DMA_STM32_PERIPH_TX_DEFAULT | DMA_STM32_PERIPH_NO_INC | DMA_STM32_MEM_INC | DMA_STM32_PERIPH_8BITS | DMA_STM32_MEM_8BITS | DMA_STM32_PRIORITY_LOW

The DMA_STM32_PERIPH_TX_DEFAULT DMA_STM32_PERIPH_RX_DEFAULT are also created with that basic default configuration.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
dts: arm: stm32 devices include the dma definitions for driver

Include the bindings macro to help configuring the dma channels
of a stm32 peripheral

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
tests: drivers: use stm32 dma driver binding macro to configure channel

Use the macro as defined by the
include/zephyr/dt-bindings/dma/stm32_dma.h to configure the
dma channel.
Use the STM32_DMA_PERIPH_TX or STM32_DMA_PERIPH_RX value.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
samples: drivers: use stm32 dma driver binding macro to configure channel

Use the macro as defined by the
include/zephyr/dt-bindings/dma/stm32_dma.h to configure the
dma channel.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

gmarull commented on include/zephyr/dt-bindings/dma/stm32_dma.h:
```diff
@@ -0,0 +1,63 @@
+ * @name custom DMA flags for channel configuration
+ * @{
+ */
+/** DMA  cyclic mode config */
+#define DMA_STM32_MODE_NORMAL			(0U << 5)
+#define DMA_STM32_MODE_CYCLIC			(1U << 5)
+
+/** DMA  transfer direction config */
+#define DMA_STM32_MEM_TO_MEM			(0U << 6)
+#define DMA_STM32_MEM_TO_PERIPH			(1U << 6)
+#define DMA_STM32_PERIPH_TO_MEM			(2U << 6)
+#define DMA_STM32_PERIPH_TO_PERIPH		(3U << 6)
+
+/** DMA  Peripheral increment Address config */
+#define DMA_STM32_PERIPH_NO_INC			(0U << 9)
+#define DMA_STM32_PERIPH_INC			(1U << 9)
+
+/** DMA  Memory increment Address config */
+#define DMA_STM32_MEM_NO_INC			(0U << 10)
+#define DMA_STM32_MEM_INC			(1U << 10)
+
+/** DMA  Peripheral data size config */
+#define DMA_STM32_PERIPH_8BITS			(0U << 11)
```
Please namespace using VND_DMA, same as https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/dt-bindings/dma/gd32_dma.h (and possibly align names, as they'll likely be the same). This is also done in a few other places, like vendor specific GPIO flags.

FRASTM committed:
include: bindings: dma macro definition for stm32 dma driver config

Defines DMA macro to help channel configuration and feature for the stm32
devices. Add one default value for most of the usecases
of peripheral DMA settings.

Signed-off-by: Francois Ramu <francois.ramu@st.com>
